src/main/java/com/hedvig/contract/customerio/EventListener.kt

```diff
@@ -244,10 +244,11 @@ class EventListener(
     eventMessage: DomainEventMessage<Any>
   ) {
     onlyRunIfEventIsNewerThanComponent(timestamp, event) {
+      val contract = contractStateRepository.findByIdOrNull(event.contractId)
       applyStrategyToContract(
         event.contractId,
         eventId = eventMessage.identifier
-      ) { it.onContractRenewalQueued(event.draftCertificate.inceptionDate!!) }
+      ) { it.onContractRenewalQueued(contract) }
```

Comment:
I prefer making the null check early
```suggestion
      ) { 
      contract?.let { contract -> 
        it.onContractRenewalQueued(contract)
         } ?: logger.error("Could not find a contract for event ContractRenewalQueued")
      }
```